core/variables.js

```diff
@@ -267,11 +268,9 @@ Blockly.Variables.getUses = function(name, workspace) {
 
 /**
  * When a variable is deleted, find and dispose of all uses of it.
- * @param {string} name Name of deleted variable.
- * @param {!Blockly.Workspace} workspace The workspace to delete uses from.
+ * @param {!Array.<!Blockly.Block>} uses An array of blocks using the variable.
  */
-Blockly.Variables.disposeUses = function(name, workspace) {
-  var uses = Blockly.Variables.getUses(name, workspace);
+Blockly.Variables.disposeUses = function(uses) {
```

Comment:
This was public in this branch, but I think it should be private.  It doesn't exist at all in code that anyone else is using--remember that variable deletion is not a concept in current blockly, so of course there's no API for it.

Erik, I'm confused by your comment on the old API.  I think what you want is just a call to Variables.delete, which does exactly what you're saying and is public.